packages/sdk/src/evm/common/verification.ts

```diff
@@ -0,0 +1,379 @@
+import invariant from "tiny-invariant";
+import { ThirdwebSDK } from "../core";
+import { allChains } from "@thirdweb-dev/chains";
+import {
+  computeDeploymentInfo,
+  encodeConstructorParamsForImplementation,
+  extractConstructorParamsFromAbi,
+  fetchAndCacheDeployMetadata,
+  fetchAndCachePublishedContractURI,
+  fetchSourceFilesFromMetadata,
+  getCreate2FactoryAddress,
+  isContractDeployed,
+  resolveContractUriFromAddress,
+} from ".";
+import { ThirdwebStorage } from "@thirdweb-dev/storage";
+import { Abi } from "../schema";
+import { ethers, utils } from "ethers";
+import { getChainProvider } from "../constants";
+import { EtherscanResult } from "../types/verification";
+import fetch from "cross-fetch";
+
+//
+// ==================================
+// ======== Core Functions ==========
+// ==================================
+//
+
+export async function isVerifiedOnEtherscan(
+  contractAddress: string,
+  api: string,
+  apiKey: string,
+): Promise<boolean> {
+  const endpoint = `${api}?module=contract&action=getsourcecode&address=${contractAddress}&apikey=${apiKey}"`;
+
+  try {
+    const result = await fetch(endpoint, {
+      method: "GET",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json;charset=UTF-8",
+      },
+    });
+
+    const data = await result.json();
+    const etherscanResult = data.result[0] as EtherscanResult;
+    if (etherscanResult.ABI === "Contract source code not verified") {
+      return false;
+    }
+    return true;
+  } catch (e) {
+    throw new Error(
+      `Error checking verification for contract ${contractAddress}: ${e}`,
+    );
+  }
+}
+
+export async function verifyThirdwebPrebuiltImplementation(
+  contractName: string,
+  chainId: number,
+  api: string,
+  apiKey: string,
+  storage: ThirdwebStorage,
+) {
+  const publishUri = await fetchAndCachePublishedContractURI(contractName);
+  const metadata = await fetchAndCacheDeployMetadata(publishUri, storage);
+  const provider = getChainProvider(chainId, {});
```

Comment:
pass `{ thirdwebApiKey: DEFAULT_API_KEY }` here to call the rpc with our default api key (might enforce it soon)